### ISSUE TITLE
Move from submodule to subtree

### ISSUE THREAD

**Issue by trolando** [utwente-fmt/ltsmin#8]
I suggest we move to use subtree instead of submodule.

We can add ".sh" files that assist updating a subtree from commit A to commit B.

Advantage
- users no longer need to manually run submodule update/init/sync
- no dependency on availability of other git repositories

Disadvantage
- maintainers need to learn about subtree method

**Comment by trolando** [comment]
Example of a .sh file that would update Sylvan:

git subtree pull --prefix=sylvan https://github.com/utwente-fmt/sylvan master --squash -m "sylvan: update to HEAD of master branch"

This updates the "sylvan" subdirectory to the latest version. --squash will mean it will squash all commits since previous subtree pull into 1 commit. It will then add a commit with the included message (-m "...") that merges the squashed commit that modifies the subtree with the checked out branch of the main repository.

**Comment by trolando** [comment]
Another advantage of this approach is that custom code in the checked out repository is better integrated with the main repository. Currently, if we customize a subrepository, this needs changes on the repository of the submodule. As far as I know, the subtree approach allows customization without modifying the submodule.

**Comment by Meijuh** [comment]
I agree, we'll need to decide whether we want this in version 2.1.1 or 2.2.

**Comment by jacopol** [comment]
Yes, a related request for the Sylvan submodule/subtree is to remove
the generated files from the repository, in particular:

modified:   configure
modified:   m4/libtool.m4
modified:   m4/ltoptions.m4
modified:   m4/ltversion.m4
modified:   tools/ltmain.sh

--
Prof.dr. J.C. van de Pol,  chair Formal Methods and Tools
Universiteit Twente, P.O. Box 217, 7500 AE  Enschede, The Netherlands
mailto:vdpol@cs.utwente.nl,    http://www.cs.utwente.nl/~vdpol
tel: +31 53 489 3017  fax: +31 53 489 3247  secr: +31 53 489 3767

**Comment by trolando** [comment]
For the related request (next time, please create separate a issue) I have prepared a commit on my personal repository, for which I created a pull request.

This removes the configure file, the aclocal.m4 file, the m4 directory and the tools directory from the sylvan tree.

**Comment by trolando** [comment]
Example of changing submodule sylvan to subtree sylvan

To remove the submodule:
```
git submodule deinit sylvan
rmdir sylvan
vim .gitmodules  # to remove sylvan
git commit -a
```

To add the subtree:
```
git remote add sylvan git@github.com:utwente-fmt/sylvan
git subtree add -P sylvan --squash sylvan/master
```

You will now find a new commit in the tree that merges the squashed sylvan commit.

**Comment by alaarman** [comment]
Sylvan is still a module?
